homeassistant/components/advantage_air/binary_sensor.py

```diff
@@ -0,0 +1,126 @@
+"""Binary Sensor platform for Advantage Air integration."""
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir motion platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    if "aircons" in instance["coordinator"].data:
+        entities = []
+        for ac_key, ac_device in instance["coordinator"].data["aircons"].items():
+            entities.append(AdvantageAirZoneFilter(instance, ac_key))
+            for zone_key, zone in ac_device["zones"].items():
+                # Only add motion sensor when motion is enabled
+                if zone["motionConfig"] == 2:
+                    entities.append(AdvantageAirZoneMotion(instance, ac_key, zone_key))
+        async_add_entities(entities)
+
+
+class AdvantageAirZoneFilter(CoordinatorEntity, BinarySensorEntity):
```

Comment:
Looks like you start to get a common ground between your entities you could break out to a base class

Reply:
I was thinking today I could have a parent for the entire integration that also handled the _ac, _zone, and device properties, however I'd likely be best to do that as a seperate PR for the entire integration.

For now I'll take your advice and make a roll up for these three like I did in climate.

Reply:
Done this, and quite like how neat it makes it. Certainly going to look at adding a parent class for the entire integration.